### PR TITLE
Task1

### PR DESCRIPTION
**Pull request by leskeg**

### CODE REVIEW

File: app/main.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
+'use strict';
+
+// App Configuration
+const API_KEY = 'c94ea59ef0324dcd827c08d3f4f96f88';
+const HOST_URL = 'https://newsapi.org/v2';
+const NEWS_PARAM = '/everything';
+const SOURCES_PARAM = '/sources';
+
+const sourcesURL = `${HOST_URL}${SOURCES_PARAM}?category=technology&language=en&apiKey=${API_KEY}`;
+const getNewsURL = (sourceName, page = 1) => `${HOST_URL}${NEWS_PARAM}?sources=${sourceName}&language=en&page=${page}&apiKey=${API_KEY}`;
```
Comment (leskeg) [inline app/main.js]: Consider to define default values always, ```sourceName = ''```, in bigger functions will give an idea of the type what need to be passed and will avoid undefined's runtime errors, it is kind of good practice

File: app/main.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        </div>
+    `;
+    return sourceItem;
+};
+
+const createNewsItem = ({author, publishedAt, title, urlToImage, url}) => {
+    // here we cannot assign default urlToImage in destructurization, as it receives null, and not undefined;
+    if (!urlToImage) {
+        urlToImage = getRandomImageUrl()
+    }
```
Comment (leskeg) [inline app/main.js]: Consider:
```javascript
const createNewsItem = ({author, publishedAt, title, urlToImage = getRandomImageUrl(), url}) => {
```
I am pretty sure will work and you will not need to write this if statement

File: app/main.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+};
+
+const renderNews = articles => {
+    articles.forEach(article => {
+        newsContainer.appendChild(createNewsItem(article));
+    });
+    document.querySelector('.show-more-button').disabled = !buttonDisabled();
+};
+
+const selectNewsSource = (sourceId) => {
```
Comment (leskeg) [inline app/main.js]: parenthesis could be removed

File: app/main.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    document.querySelector('.show-more-button').classList.remove('hidden');
+    fetch(getNewsURL(sourceId), { mode: 'cors' })
+        .then(res => res.json())
+        .then(({articles}) => renderNews(articles))
+        .catch(error => console.log(error));
+};
+
+const getPage = () => (newsContainer.children.length / 20) + 1;
+
+const buttonDisabled = () =>
```
Comment (leskeg) [inline app/main.js]: Maybe no need page const on this function
```javascript
const buttonDisabled = () => getPage() === Math.ceil(page);

File: app/main.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        newsContainer.appendChild(createNewsItem(article));
+    });
+    document.querySelector('.show-more-button').disabled = !buttonDisabled();
+};
+
+const selectNewsSource = (sourceId) => {
+    sourcesContainer.classList.add('hidden');
+    newsContainer.innerHTML = '';
+    sourcesSelector.value = sourceId;
+    document.querySelector('.show-more-button').classList.remove('hidden');
```
Comment (leskeg) [inline app/main.js]: You could declare selectors separately and reference them instead of using values directly

File: app/main.js (hunk 2 above)
Comment (aliaksei-lyzo) [reply app/main.js]: I totally thought about that, but default assign works if the parameter is undefined, here API returns null if no image exists, so default assignment doesn't work.... :(

File: app/main.js (hunk 3 above)
Comment (aliaksei-lyzo) [reply app/main.js]: Fixed

File: app/main.js (hunk 5 above)
Comment (aliaksei-lyzo) [reply app/main.js]: Fixed. Did that with others, don't know why I missed the button :)

File: app/main.js (hunk 4 above)
Comment (aliaksei-lyzo) [reply app/main.js]: Totally cool! I wanted to add some more logic, stopped on this variant and forgot to clean up :) Thanks!

File: app/main.js (hunk 1 above)
Comment (aliaksei-lyzo) [reply app/main.js]: Gr8! Will adopt such practice, looks really convenient.

**Pull request merged**